src/ConnectedMode/QualityProfiles/QualityProfileUpdater.cs

```diff
@@ -72,13 +72,18 @@ public async Task UpdateAsync()
 
             try
             {
-                // TODO: only raise the event if at least one QP actually changed...
-                await runner.RunAsync(token => qualityProfileDownloader.UpdateAsync(config.Project, null, CancellationToken.None));
-                QualityProfilesChanged?.Invoke(this, EventArgs.Empty);
+                await runner.RunAsync(async token =>
+                {
+                    if (await qualityProfileDownloader.UpdateAsync(config.Project, null, token))
+                    {
+                        QualityProfilesChanged?.Invoke(this, EventArgs.Empty);
+                    }
+                });
             }
-            catch (OperationCanceledException)
+            catch (Exception e) when (e is OperationCanceledException || e is InvalidOperationException)
             {
                 // no-op - job was cancelled
+                logger.LogVerbose(e.ToString());
```

Comment:
Prefix the string with the feature name to give a bit more context: `[QualityProfiles] `